### PR TITLE
Add ref/tests for Span-based Encoding APIs

### PR DESCRIPTION
**Pull request by stephentoub**
cc: @KrzysztofCwalina, @tarekgh 
Fixes https://github.com/dotnet/corefx/issues/22431
Depends on https://github.com/dotnet/coreclr/pull/13269

### CODE REVIEW

**Comment by tarekgh** [comment]
LGTM too. thanks again.

File: src/System.Private.Xml/src/System/Xml/XmlEncoding.cs [hunk 1]
```diff
@@ -331,10 +334,14 @@ public override byte[] GetPreamble()
         {
             return new byte[4] { 0x00, 0x00, 0xfe, 0xff };
         }
+
+        private ReadOnlySpan<byte> Preamble => s_preamble;
```
Comment (justinvp) [inline src/System.Private.Xml/src/System/Xml/XmlEncoding.cs]: `public override`?

Comment (stephentoub) [inline src/System.Private.Xml/src/System/Xml/XmlEncoding.cs]: Ugh. Yes. Fixing.

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@dotnet-bot test Packaging All Configurations x64 Debug Build please
@dotnet-bot test UWP CoreCLR x64 Debug Build please
@dotnet-bot test UWP NETNative x86 Release Build please

**Comment by tarekgh** [comment]
LGTM. I assume the implementation of the added APIs already mirrored to corert to ensure .NetNative will not fail.

**Comment by stephentoub** [comment]
> I assume the implementation of the added APIs already mirrored to corert to ensure .NetNative will not fail.

Yes, mirrored to corert in https://github.com/dotnet/corert/pull/4302, but still waiting for those changes to be propagated to be available here.

**Comment by stephentoub** [comment]
@dotnet-bot test Packaging All Configurations x64 Debug Build please
@dotnet-bot test UWP NETNative x86 Release Build please

**Comment by stephentoub** [comment]
@dotnet-bot test Packaging All Configurations x64 Debug Build please
@dotnet-bot test UWP NETNative x86 Release Build please

**Pull request merged**